### PR TITLE
Add warning on pillow version if reading a MPO image

### LINKED ISSUE #2445: Segfault caused by skimage.io.imread on MPO files
## Description
I'm running caffe on millions of images and one certain image caused segmentation fault and aborted the whole process while being read by skimage.io.

Code gist:
```python
import skimage.io
try:
  im = skimage.io.imread(fname)
  ...
except:
  ...
```
Error message:
```
*** Error in `python`: free(): invalid size: 0x000000000b5cb340 ***
```

Here is this image https://cloud.githubusercontent.com/assets/13360214/21850830/6aadccd6-d847-11e6-84d6-e6e955592957.jpg . Reading it through `cv2` works fine but `skimage.io` causes segmentation fault.

Environments:
- ubuntu 12.04
- python 2.7.11, anaconda 2.2.0
- skimage 0.12.3

The backtrace of reading this image by `skimage.io.imread`:
```
GNU gdb (Ubuntu/Linaro 7.4-2012.04-0ubuntu2.1) 7.4-2012.04
Copyright (C) 2012 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
For bug reporting instructions, please see:
<http://bugs.launchpad.net/gdb-linaro/>...
Reading symbols from /home/username/anaconda/bin/python...done.
(gdb) run -c 'import skimage.io; skimage.io.imread("segfault.jpg")'
Starting program: /home/username/anaconda/bin/python -c 'import skimage.io; skimage.io.imread("segfault.jpg")'
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffedf40700 (LWP 30490)]
[New Thread 0x7fffed73f700 (LWP 30491)]
[New Thread 0x7fffeaf3e700 (LWP 30492)]

Program received signal SIGSEGV, Segmentation fault.
0x00007fffe22a2d47 in decompress_onepass (cinfo=0x1901f70, output_buf=0x18e51e8) at jdcoefct.c:188
188	jdcoefct.c: No such file or directory.
(gdb) bt
#0  0x00007fffe22a2d47 in decompress_onepass (cinfo=0x1901f70, output_buf=0x18e51e8) at jdcoefct.c:188
#1  0x00007fffe22a822e in process_data_simple_main (cinfo=0x1901f70, output_buf=0x7fffd776f4c8, out_row_ctr=0x7fffffffc3bc, out_rows_avail=1) at jdmainct.c:355
#2  0x00007fffe22a071a in jpeg_read_scanlines (cinfo=cinfo@entry=0x1901f70, scanlines=scanlines@entry=0x7fffd776f4c8, max_lines=max_lines@entry=1) at jdapistd.c:174
#3  0x00007fffe2508380 in ImagingJpegDecode (im=0x18bb530, state=0x7fffd776f490, buf=0x1923074 "", bytes=65536) at libImaging/JpegDecode.c:232
#4  0x00007fffe24e3cb5 in _decode (decoder=0x7fffd776f470, args=<optimized out>) at decode.c:130
#5  0x00007ffff7aefe15 in call_function (oparg=<optimized out>, pp_stack=0x7fffffffc588) at Python/ceval.c:4350
#6  PyEval_EvalFrameEx (f=<optimized out>, throwflag=<optimized out>) at Python/ceval.c:2987
#7  0x00007ffff7aefb95 in fast_function (nk=<optimized out>, na=<optimized out>, n=<optimized out>, pp_stack=0x7fffffffc6f8, func=0x7fffd77c3578) at Python/ceval.c:4436
#8  call_function (oparg=<optimized out>, pp_stack=0x7fffffffc6f8) at Python/ceval.c:4371
#9  PyEval_EvalFrameEx (f=<optimized out>, throwflag=<optimized out>) at Python/ceval.c:2987
#10 0x00007ffff7af0a2e in PyEval_EvalCodeEx (co=0x7fffe27561b0, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=1, kws=0x7fffd77dded8, kwcount=0, defs=0x7fffe175d328, 
    defcount=1, closure=0x0) at Python/ceval.c:3582
#11 0x00007ffff7aefa55 in fast_function (nk=<optimized out>, na=1, n=<optimized out>, pp_stack=0x7fffffffc918, func=0x7fffe1760c80) at Python/ceval.c:4446
#12 call_function (oparg=<optimized out>, pp_stack=0x7fffffffc918) at Python/ceval.c:4371
#13 PyEval_EvalFrameEx (f=<optimized out>, throwflag=<optimized out>) at Python/ceval.c:2987
#14 0x00007ffff7af0a2e in PyEval_EvalCodeEx (co=0x7fffe27bbeb0, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=1, kws=0x0, kwcount=0, defs=0x0, defcount=0, closure=0x0)
    at Python/ceval.c:3582
#15 0x00007ffff7a6c3a1 in function_call (func=0x7fffe1760b18, arg=0x7fffd77be2d0, kw=0x0) at Objects/funcobject.c:526
#16 0x00007ffff7a3cd23 in PyObject_Call (func=0x7fffe1760b18, arg=<optimized out>, kw=<optimized out>) at Objects/abstract.c:2546
#17 0x00007ffff7a3ffec in call_function_tail (args=0x7fffd77be2d0, callable=0x7fffe1760b18) at Objects/abstract.c:2578
#18 PyObject_CallFunction (callable=0x7fffe1760b18, format=<optimized out>) at Objects/abstract.c:2602
#19 0x00007ffff7a87c1b in _PyObject_GenericGetAttrWithDict (obj=0x7fffd788f8d0, name=0x7ffff7e9fab0, dict=0x0) at Objects/object.c:1439
#20 0x00007ffff08bfc04 in PyArray_GetAttrString_SuppressException (obj=obj@entry=0x7fffd788f8d0, name=name@entry=0x7ffff09d0c08 "__array_interface__") at numpy/core/src/multiarray/common.c:76
#21 0x00007ffff08cb0db in PyArray_FromInterface (origin=origin@entry=0x7fffd788f8d0) at numpy/core/src/multiarray/ctors.c:2170
#22 0x00007ffff08cda01 in PyArray_GetArrayParamsFromObject (context=<optimized out>, out_arr=0x7fffffffcf70, out_dims=0x7fffffffcf90, out_ndim=0x7fffffffcf40, out_dtype=0x7fffffffcf80, 
    writeable=0 '\000', requested_dtype=0x0, op=<optimized out>) at numpy/core/src/multiarray/ctors.c:1495
#23 PyArray_FromAny (op=op@entry=0x7fffd788f8d0, newtype=newtype@entry=0x0, min_depth=0, max_depth=0, flags=flags@entry=112, context=0x0) at numpy/core/src/multiarray/ctors.c:1687
#24 0x00007ffff08cddb5 in PyArray_CheckFromAny (op=0x7fffd788f8d0, descr=0x0, min_depth=min_depth@entry=0, max_depth=max_depth@entry=0, requires=112, context=context@entry=0x0)
    at numpy/core/src/multiarray/ctors.c:1865
#25 0x00007ffff095d76d in _array_fromobject (__NPY_UNUSED_TAGGEDignored=<optimized out>, args=<optimized out>, kws=0x7fffd7841168) at numpy/core/src/multiarray/multiarraymodule.c:1714
#26 0x00007ffff7aef747 in do_call (nk=<optimized out>, na=<optimized out>, pp_stack=0x7fffffffd2e8, func=0x7ffff7e3e5a8) at Python/ceval.c:4565
#27 call_function (oparg=<optimized out>, pp_stack=0x7fffffffd2e8) at Python/ceval.c:4373
#28 PyEval_EvalFrameEx (f=<optimized out>, throwflag=<optimized out>) at Python/ceval.c:2987
#29 0x00007ffff7af0a2e in PyEval_EvalCodeEx (co=0x7fffdbac1bb0, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=1, kws=0xbed980, kwcount=2, defs=0x7fffdbaac458, 
    defcount=2, closure=0x0) at Python/ceval.c:3582
#30 0x00007ffff7aefa55 in fast_function (nk=<optimized out>, na=1, n=<optimized out>, pp_stack=0x7fffffffd508, func=0x7fffdb660a28) at Python/ceval.c:4446
#31 call_function (oparg=<optimized out>, pp_stack=0x7fffffffd508) at Python/ceval.c:4371
#32 PyEval_EvalFrameEx (f=<optimized out>, throwflag=<optimized out>) at Python/ceval.c:2987
#33 0x00007ffff7af0a2e in PyEval_EvalCodeEx (co=0x7fffdbabcc30, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=1, kws=0x7ffff7f97068, kwcount=0, defs=0x7fffdbaac4a0, 
    defcount=2, closure=0x0) at Python/ceval.c:3582
#34 0x00007ffff7a6c4a8 in function_call (func=0x7fffdb6609b0, arg=0x7fffe178f050, kw=0x7fffd783aa28) at Objects/funcobject.c:526
#35 0x00007ffff7a3cd23 in PyObject_Call (func=0x7fffdb6609b0, arg=<optimized out>, kw=<optimized out>) at Objects/abstract.c:2546
#36 0x00007ffff7aed797 in ext_do_call (nk=-512167856, na=<optimized out>, flags=<optimized out>, pp_stack=0x7fffffffd7f8, func=0x7fffdb6609b0) at Python/ceval.c:4663
#37 PyEval_EvalFrameEx (f=<optimized out>, throwflag=<optimized out>) at Python/ceval.c:3026
#38 0x00007ffff7af0a2e in PyEval_EvalCodeEx (co=0x7fffe2796830, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=2, kws=0x7fffd785be78, kwcount=1, defs=0x0, defcount=0, 
    closure=0x0) at Python/ceval.c:3582
#39 0x00007ffff7a6c4a8 in function_call (func=0x7fffe12b7ed8, arg=0x7fffdbaac4d0, kw=0x7fffd77d6d70) at Objects/funcobject.c:526
#40 0x00007ffff7a3cd23 in PyObject_Call (func=0x7fffe12b7ed8, arg=<optimized out>, kw=<optimized out>) at Objects/abstract.c:2546
#41 0x00007ffff7aed797 in ext_do_call (nk=-609565488, na=<optimized out>, flags=<optimized out>, pp_stack=0x7fffffffdae8, func=0x7fffe12b7ed8) at Python/ceval.c:4663
#42 PyEval_EvalFrameEx (f=<optimized out>, throwflag=<optimized out>) at Python/ceval.c:3026
#43 0x00007ffff7af0a2e in PyEval_EvalCodeEx (co=0x7fffe27adab0, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=1, kws=0x7ffff7fc0da8, kwcount=0, defs=0x7fffdbad6c98, 
    defcount=3, closure=0x0) at Python/ceval.c:3582
#44 0x00007ffff7aefa55 in fast_function (nk=<optimized out>, na=1, n=<optimized out>, pp_stack=0x7fffffffdd08, func=0x7fffdb6616e0) at Python/ceval.c:4446
#45 call_function (oparg=<optimized out>, pp_stack=0x7fffffffdd08) at Python/ceval.c:4371
#46 PyEval_EvalFrameEx (f=<optimized out>, throwflag=<optimized out>) at Python/ceval.c:2987
#47 0x00007ffff7af0a2e in PyEval_EvalCodeEx (co=0x7ffff7ecfab0, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=0, kws=0x0, kwcount=0, defs=0x0, defcount=0, closure=0x0)
    at Python/ceval.c:3582
---Type <return> to continue, or q <return> to quit---
#48 0x00007ffff7af0b42 in PyEval_EvalCode (co=<optimized out>, globals=<optimized out>, locals=<optimized out>) at Python/ceval.c:669
#49 0x00007ffff7b0f5cc in run_mod (arena=0x645f50, flags=<optimized out>, locals=0x7ffff7f6c168, globals=0x7ffff7f6c168, filename=0x7ffff7b3f7ed "<string>", mod=<optimized out>) at Python/pythonrun.c:1370
#50 PyRun_StringFlags (str=0x601010 "import skimage.io; skimage.io.imread(\"segfault.jpg\")\n", start=257, globals=0x7ffff7f6c168, locals=0x7ffff7f6c168, flags=<optimized out>) at Python/pythonrun.c:1333
#51 0x00007ffff7b108f0 in PyRun_SimpleStringFlags (command=0x601010 "import skimage.io; skimage.io.imread(\"segfault.jpg\")\n", flags=0x7fffffffdf90) at Python/pythonrun.c:974
#52 0x00007ffff7b26457 in Py_Main (argc=<optimized out>, argv=<optimized out>) at Modules/main.c:589
#53 0x00007ffff6d3676d in __libc_start_main () from /lib/x86_64-linux-gnu/libc.so.6
#54 0x0000000000400649 in _start ()
```

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Fixes #2445 based on @sciunto [comment](https://github.com/scikit-image/scikit-image/issues/2445#issuecomment-543537856).

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/io/_plugins/pil_plugin.py [hunk 1]
```diff
@@ -36,6 +37,10 @@ def imread(fname, dtype=None, img_num=None, **kwargs):
             return pil_to_ndarray(im, dtype=dtype, img_num=img_num)
     else:
         im = Image.open(fname)
+        if im.format == 'MPO' and Image.__version__ < '6.0.0':
+            warnings.warn("You are trying to read a MPO image. "
+                          "To ensure a good support of this format, "
+                          "please upgrade pillow to 6.0.0 version.")
```
Comment (sciunto) [inline skimage/io/_plugins/pil_plugin.py]: Could you add a stacklevel please?

Comment (sciunto) [inline skimage/io/_plugins/pil_plugin.py]:
```suggestion
                          "please upgrade pillow to 6.0.0 version or later.")
```

**Comment by rfezzani** [comment]
You are absolutely right @hmaarrfk, I don't know how I forgot that :confused:... Thank you!

File: skimage/io/_plugins/pil_plugin.py [hunk 2]
```diff
@@ -36,6 +38,12 @@ def imread(fname, dtype=None, img_num=None, **kwargs):
             return pil_to_ndarray(im, dtype=dtype, img_num=img_num)
     else:
         im = Image.open(fname)
+        pil_version = LooseVersion(Image.__version__)
+        if im.format == 'MPO' and pil_version < LooseVersion('6.0.0'):
```
Comment (hmaarrfk) [inline skimage/io/_plugins/pil_plugin.py]:
```suggestion
        if im.format == 'MPO' and pil_version < '6.0.0':
```

```python

In [1]:  from distutils.version import LooseVersion                             

In [2]: LooseVersion('1.0') < '6.0.0'                                           
Out[2]: True

In [3]: LooseVersion('1.0') > '0.9.1'                                           
Out[3]: True

In [4]:  

```

**Comment by rfezzani** [comment]
Thank you both for your super reactivity ;-)

**Comment by sciunto** [comment]
good point @hmaarrfk !

File: skimage/io/_plugins/pil_plugin.py [hunk 3]
```diff
@@ -36,6 +38,12 @@ def imread(fname, dtype=None, img_num=None, **kwargs):
             return pil_to_ndarray(im, dtype=dtype, img_num=img_num)
     else:
         im = Image.open(fname)
+        pil_version = LooseVersion(Image.__version__)
```
Comment (sciunto) [inline skimage/io/_plugins/pil_plugin.py]: From travis: Looks like Image.__version__ exists only in the latest versions...

**Comment by emmanuelle** [comment]
Thank you, merging!

**Pull request merged**